kernelci/config/rootfs.py

```diff
@@ -93,13 +94,22 @@ class RootFSFactory(YAMLObject):
 
     @classmethod
     def from_yaml(cls, name, rootfs):
-        rootfs_type = rootfs.get('rootfs_type')
-        kw = {
-            'name': name,
-            'rootfs_type': rootfs_type,
-        }
-        rootfs_cls = cls._rootfs_types[rootfs_type] if rootfs_type else RootFS
-        return rootfs_cls.from_yaml(rootfs, kw)
+        try:
+            rootfs_type = rootfs.get('rootfs_type')
+            if rootfs_type is None:
+                raise TypeError("rootfs_type cannot be Empty")
+            elif rootfs_type not in cls._rootfs_types:
+                raise ValueError("Unsupported value {}".format(rootfs_type))
+            else:
+                kw = {
+                    'name': name,
+                    'rootfs_type': rootfs_type,
+                }
+                rootfs_cls = cls._rootfs_types[rootfs_type] if rootfs_type \
+                    else RootFS
+                return rootfs_cls.from_yaml(rootfs, kw)
+        except (TypeError, ValueError) as e:
```

Comment:
I don't see any reason why trying to catch exceptions if all we're doing is raising another one to abort the command.  I think we should just not have a try-except block here, this looks like diverting exception handling into flow control.

Reply:
okay removed try-except block - just used `raise`